### ISSUE TITLE
[Bug] Opening drowdown after focus on other field.

### ISSUE THREAD

**Issue by Danaru87** [GiancarloCode/form_bloc#28]
When you have a Dropdown,

if you focus another field, when coming back to list will open => close very fast the list and close => open the keyboard giving focus to the field. 

Seems to be a bug of https://github.com/adee42/flutter_keyboard_visibility, onChange never fired.

in dropdown_field_bloc_builder.dart, lines 268 - 275, changing:

```dart
if (_keyboardVisibility.isKeyboardVisible) {
        //_effectiveFocusNode.requestFocus();
        await Future<void>.delayed(Duration(milliseconds: 1));
        _effectiveFocusNode.unfocus();
        await Future<void>.delayed(Duration(
            milliseconds:
                widget.millisecondsForShowDropdownItemsWhenKeyboardIsOpen));
      }
```

by
```dart
 if (MediaQuery.of(context).viewInsets.bottom != 0) {
        FocusScope.of(context).requestFocus(new FocusNode());
        await Future.delayed(Duration(milliseconds: 1));
      }
```
solved this bug.
Solution based on comments from the link into your TODO comment. (line 267)

https://github.com/flutter/flutter/issues/18672#issuecomment-426522889

**Comment by jawadsahil** [comment]
I just came across this issue, and I was thinking to post the issue :) But it is already here ... Thanks

**Comment by GiancarloCode** [comment]
@Danaru87 Thanks for reporting this

`MediaQuery.of(context).viewInsets.bottom != 0` Always return `false` And `flutter_keyboard_visibility` not work :(.

I need a way to detect if the keyboard is visible but I can't find any that works for now, if you get a way to do it let me know and fix it. I will try to look at this again the day after tomorrow.

**Comment by Danaru87** [comment]
Strange behavior, using MediaQuery.of(context).viewInsets.bottom != 0 worked for me.
Maybe you use a context without SafeArea.

This plugin work to detect keyboard opened or closed: https://github.com/IsaiasSantana/keyboard_utils


But, whith dropdown opened, if you tap outside the dropdown, focus stay on it and dont close option list.

I'm working on a solution.

**Comment by Danaru87** [comment]
PR ready #32 .

The code is not super clean, if you have any suggestions I can make the necessary changes.

For those who can't wait for a new release, you can refer my repository into your pubspec.yaml:

```yaml
  flutter_form_bloc:
    git:
      url: https://github.com/UPrefer/form_bloc.git
      path: packages/flutter_form_bloc
```

**Comment by GiancarloCode** [comment]
Fixed in form_bloc 0.10.4.
Thank you @Danaru87

**Comment by GiancarloCode** [comment]
P.D. @Danaru87 @jawadsahil  Check pinned issues
